bot/seasons/evergreen/bookmark.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+import discord
+from discord.ext import commands
+
+log = logging.getLogger(__name__)
+
+
+class Bookmark(commands.Cog):
+    """A cog for Bookmarking a message."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    @commands.command(name="bookmark", aliases=("bm", "pin"))
+    async def bookmark(self, ctx: commands.Context, jump_url: str = None, *args) -> None:
+        """Bookmarks a message."""
+        if jump_url is not None and jump_url != "*":
+            if "discordapp.com/channels" not in jump_url:
+                await ctx.send("I can't find the associated message,"
+                               " the command support the following syntax:\n"
+                               " `.bm [message url] (bookmark name)`")
+                return
+        if jump_url is None or jump_url == "*":
+            channel = ctx.channel
+            async for x in channel.history(limit=2):
+                message_id = x.id

```

Comment:
You can simplify it to
```py
Change it to
async for message in channel.history(limit=2):
  jump_url = message.jump_url
 ```

Reply:
Thanks <3 , forgot to mention that I am using this now.